SimG4Core/Application/python/g4SimHits_cfi.py

```diff
@@ -45,6 +45,9 @@
     DoFineCalo = cms.bool(False),
     # currently unused; left in place for future studies
     EminFineTrack = cms.double(10000.0),
+    FineCaloNames = cms.vstring('ECAL', 'HCAL', 'HGCal', 'HFNoseVol', 'VCAL'),
+    FineCaloLevels = cms.vint32(4, 4, 8, 3, 3),
```

Comment:
is it clear that there will always be only level per subdetector?

Reply:
Kevin, you are right. They cannot be constants - they depend on geometry implementation. But  that is rare. So I keep them in parameter statements but more stable than the choice of FineCalo in an application

Reply:
That's a good point, but my intended question was slightly different: what if HGCal needs two levels, e.g. 8 and 10? Will that ever happen, or are the sensitive elements only ever at one level for a subdetector?

Reply:
We refer to only top level volumes and not sensitive detector. These top volumes so far happens to be in one level. If not, the same name may happen multiple times with the corresponding levels and then the current logic will work.